### ISSUE TITLE
detection is too narrow and flat angles prohibit detection

### ISSUE THREAD

**Issue by 1-rafael-1** [1-rafael-1/simple-robot#2]
the bot hits obstacles with its tracks because detection has no sweep and only 15° angle. Need a better suited sensor or a second hc-sr04 or mount the hc-sr04 on something that can sweep...

**Comment by XiaoPengYouCode** [comment]
try to add the sensor on an servo motor and rotate it time and time again(just like a small Lidar sensor)

**Comment by 1-rafael-1** [comment]
> try to add the sensor on an servo motor and rotate it time and time again(just like a small Lidar sensor)

Yes that is one option I can follow. I believe I will maybe do this to explore how spatial awareness and pathfinding works. For starters I will try a IR sensor that has 35° angle. That should be better for all sorts of obstacles because sound will bounce but light will scatter and so this should work better.

**Comment by XiaoPengYouCode** [comment]
how about try to use a cheap off-the-shelf LiDAR, it can map environment srounded into a 2D point-cloud-data, and you can use it to build automatic cruise.

**Comment by XiaoPengYouCode** [comment]
> how about try to use a cheap off-the-shelf LiDAR, it can map environment srounded into a 2D point-cloud-data, and you can use it to build automatic cruise.

that's more cool maybe for your kids❤️
